woocommerce/woocommerce-android
# Order Creation: remove product from the order

Closes: #5557 


Please merge after #5563

### Description
This PR handles the removal of products from the Order.
Please note that we don't have a finalized design for this, so I followed the design mentioned on this comment p91TBi-4zK-p2#comment-3547, which the iOS team did on their PR https://github.com/woocommerce/woocommerce-ios/pull/5710 

### Testing instructions
1. Apply the patch below.
2. Open the order list screen, and click on the FAB button.
3. Click on one of the products.
4. Notice the modal screen.
5. Click on Remove product.
6. Confirm that the item is removed.

### Images/gif
<img src="https://user-images.githubusercontent.com/1657201/147491979-01fee64e-c5e2-4aea-a2e4-1191b4cf8245.png" width=400/>

### Patch
```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(revision 658757aa9382a327c3db81d0aead25e66c09bbba)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(date 1640694086896)
@@ -16,6 +16,7 @@
 import com.woocommerce.android.viewmodel.ScopedViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.withContext
+import java.math.BigDecimal
 import javax.inject.Inject
 
 @HiltViewModel
@@ -57,6 +58,49 @@
         orderDraft = orderDraft.copy(
             currency = parameterRepository.getParameters(PARAMETERS_KEY, savedState).currencyCode.orEmpty()
         )
+        updateOrderItems(
+            listOf(
+                Order.Item(
+                    0L,
+                    productId = 44,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU123",
+                    quantity = 1f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 45,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU124",
+                    quantity = 1f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 47,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU125",
+                    quantity = 1f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                )
+            )
+        )
     }
 
     fun onOrderStatusChanged(status: Order.Status) {
```
